app/src/main/java/org/p2p/wallet/sell/ui/extensions/SellTransactionDetailsBindingExtensions.kt

```diff
@@ -0,0 +1,181 @@
+            bodyIconRes = R.drawable.ic_alert_rounded
+            bodyIconTint = R.color.icons_rose
+            buttonTitle = getString(R.string.common_try_again)
+            action = SellTransactionDetailsButtonAction.TRY_AGAIN
+        }
+        MoonpaySellTransaction.SellTransactionStatus.UNKNOWN -> {
+            Timber.e(IllegalArgumentException("Can't render unknown transaction"))
+            return
+        }
+    }
+
+    renderTitleAndBody(
+        title = title,
+        body = body,
+        bodyIcon = bodyIconRes,
+        bodyTextColorRes = bodyTextColorRes,
+        bodyIconTint = ColorStateList.valueOf(getColor(bodyIconTint)),
+        bodyBackground = bodyBackground
+    )
+
+    renderButtons(
+        action = action,
+        buttonTitle = buttonTitle,
+        status = details.status,
+        onActionButtonClicked = onActionButtonClicked,
+        onRemoveClicked = onRemoveClicked
+    )
+}
+
+private fun LayoutSellTransactionDetailsBinding.renderTitleAndBody(
+    title: String,
+    body: CharSequence, // can be spanned link
+    bodyIcon: Int,
+    bodyIconTint: ColorStateList,
+    bodyBackground: Int,
+    bodyTextColorRes: Int
+) {
+    textViewTitle.text = title
+    textViewMessageBody.text = body
+    textViewMessageBody.setTextColorRes(bodyTextColorRes)
+    textViewMessageBody.setLinkTextColor(getColor(R.color.text_sky))
+    imageViewMessageIcon.setImageResource(bodyIcon)
+    imageViewMessageIcon.imageTintList = bodyIconTint
+
+    containerMessage.setBackgroundResource(bodyBackground)
+}
+
+private fun LayoutSellTransactionDetailsBinding.renderButtons(
+    action: SellTransactionDetailsButtonAction,
+    buttonTitle: String,
+    status: MoonpaySellTransaction.SellTransactionStatus,
+    onActionButtonClicked: (SellTransactionDetailsButtonAction) -> Unit,
+    onRemoveClicked: () -> Unit
+) {
+    buttonAction.text = buttonTitle
+    buttonAction.setOnClickListener { onActionButtonClicked(action) }
+    buttonRemove.isVisible =
+        status == MoonpaySellTransaction.SellTransactionStatus.WAITING_FOR_DEPOSIT ||
+        status == MoonpaySellTransaction.SellTransactionStatus.FAILED
+    buttonRemove.setOnClickListener { onRemoveClicked() }
+}
+
+private fun LayoutSellTransactionDetailsBinding.renderAmounts(
+    solAmount: String,
+    usdAmount: String,
+    receiverAddress: String,
+    isAddressSolWallet: Boolean
+) {
+    textViewAmount.text = "$solAmount $SOL_SYMBOL"
+    textViewUsdValue.text = getString(R.string.sell_lock_usd_amount, usdAmount)
+
+    textViewReceiverAddress.text = receiverAddress.let { if (isAddressSolWallet) it.cutMiddle() else it }
+}
+
+private fun LayoutSellTransactionDetailsBinding.renderCopyButton(
+    isAddressSolWallet: Boolean,
+    walletAddress: String,
+    onAddressCopied: () -> Unit,
+) {
+    imageViewCopy.isVisible = isAddressSolWallet
+    imageViewCopy.setOnClickListener {
+        context.copyToClipBoard(walletAddress)
+        onAddressCopied()
+    }
+}
```

Comment:
Imho, the UI rendering should be in fragment. 
It's not about static extensions.
The fragment which uses these extensions are absolutely empty, as if it has empty space for exactly this code

@kabramovp2p what do you think?

Reply:
I agree to make UI related logic in fragment

Reply:
how can i use the same render logic in another fragment?